cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -103,6 +103,12 @@ struct ARROW_EXPORT S3Options {
   /// server).
   std::string region;
 
+  /// if unset (or any value less than 0), the AWS SDK default values will be used.
+  /// request_timeout default 3 seconds
+  /// connect_timeout default 1 second
+  double request_timeout = -1;
+  double connect_timeout = -1;
```

Comment:
```suggestion
  /// Socket read timeout, the default is 3 seconds.
  /// If less than zero then this property is ignored and the default is used.
  /// If zero then ???
  double request_timeout = -1;
  /// Socket connect timeout, the default is 1 second.
  /// If less than zero then this property is ignored and the default is used.
  /// If zero then ???
  double connect_timeout = -1;
```

I don't think we need to give as much detail as we do in python (although maybe we should) but we should be unambiguous.  Note that this comment is used to automatically generate this API doc: https://arrow.apache.org/docs/cpp/api/filesystem.html#_CPPv4N5arrow2fs9S3OptionsE

So even though there is a bit of repetition it helps to have a comment for each field and not to bunch them up like this.